### PR TITLE
Enable Genymotion debug

### PR DESCRIPTION
**Pull request by jonathanmos**
## Description
Enable running debug tests on Genycloud (discussed in [this](https://github.com/wix/Detox/issues/2655) issue)

### CODE REVIEW

File: detox/test/e2e/setup.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const { device } = require('detox');
 
 beforeAll(async () => {
+  await device.reverseTcpPort(8081);
```
Comment (noomorph) [inline detox/test/e2e/setup.js:4]: Is that port of the Metro bundler?

File: detox/android/detox/src/full/java/com/wix/detox/Detox.java [hunk 2]
```diff
@@ -72,6 +74,7 @@
 public final class Detox {
     private static final String INTENT_LAUNCH_ARGS_KEY = "launchArgs";
     private static final long ACTIVITY_LAUNCH_TIMEOUT = 10000L;
+    private static final String DEBUG_HOST_KEY = "debug_http_host";
```
Comment (d4vidi) [inline detox/android/detox/src/full/java/com/wix/detox/Detox.java:77]:
```suggestion
    private static final String RN_DEBUG_HOST_KEY = "debug_http_host";
```

**Review by d4vidi — CHANGES_REQUESTED** [review]
I'm not mentally ready to do this world-wide, as implicated by this PR.
I think we should first introduce a more local solution:
1. Specify the preferences tweak in our internal project's `DetoxTest.java`
2. Call `device.reverseTcpPort(8081)` in the single test suite where it really matters, in the associated project.

**Comment by d4vidi** [comment]
@jonathanmos I think we should reiterate this, and decide on one of these 2 options:
1. Avoid integrating into Detox at this point, and solve this locally (as I've previously suggested).
2. Introduce this into Detox in a more generic and holistic way. For example, instead of `isGenymotion`, we should allow for accepting an argument that holds the URL to override with (e.g. `localhost:8081`). Also, if we do this, then we'd also have to introduce a test scheme that would run in Debug and verify that this thing works continuously.

**Comment by jonathanmos** [comment]
Will be solved locally

**Pull request closed without merging**